library/src/main/java/com/github/pwittchen/reactivenetwork/library/ReactiveNetwork.java

```diff
@@ -117,23 +132,42 @@ public static ReactiveNetwork create() {
    */
   public static Observable<Boolean> observeInternetConnectivity(final int intervalInMs,
       final String host, final int port, final int timeoutInMs) {
+    return observeInternetConnectivity(intervalInMs, intervalInMs, host, port, timeoutInMs);
+  }
+
+  /**
+   * Observes connectivity with the Internet by opening socket connection with remote host
+   *
+   * @param initialIntervalInMs in milliseconds determining the delay of the first connectivity check
+   * @param intervalInMs in milliseconds determining how often we want to check connectivity
+   * @param host for checking Internet connectivity
+   * @param port for checking Internet connectivity
+   * @param timeoutInMs for pinging remote host in milliseconds
+   * @return RxJava Observable with Boolean - true, when we have connection with host and false if
+   * not
+   */
+  public static Observable<Boolean> observeInternetConnectivity(final int initialIntervalInMs, final int intervalInMs,
+                                                                final String host, final int port, final int timeoutInMs) {
+    if (initialIntervalInMs < 0) {
+      throw new IllegalArgumentException("initialIntervalInMs is not a non-zero number");
+    }
     Preconditions.checkPositive(intervalInMs, "intervalInMs is not positive number");
     Preconditions.checkNotNullOrEmpty(host, "host is null or empty");
     Preconditions.checkPositive(port, "port is not positive number");
     Preconditions.checkPositive(timeoutInMs, "timeoutInMs is not positive number");
 
-    return Observable.interval(intervalInMs, TimeUnit.MILLISECONDS, Schedulers.io())
-        .map(new Func1<Long, Boolean>() {
-          @Override public Boolean call(Long tick) {
-            try {
-              Socket socket = new Socket();
-              socket.connect(new InetSocketAddress(host, port), timeoutInMs);
-              return socket.isConnected();
-            } catch (IOException e) {
-              return Boolean.FALSE;
-            }
-          }
-        })
-        .distinctUntilChanged();
+    return Observable.interval(initialIntervalInMs, intervalInMs, TimeUnit.MILLISECONDS, Schedulers.io())
```

Comment:
Please, remember to update documentation in `README.md` file when you are updating library API.